Samples/HelloWorld/OrleansHost/Program.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace OrleansHost
+{
+    /// <summary>
+    /// Orleans test host
+    /// </summary>
+    public class Program
+    {
+        private static OrleansHostWrapper hostWrapper;
+        static int Main(string[] args)
+        {
+            int returnCode = StartSilo(args);
+
+            Console.WriteLine("Press Enter to terminate...");
```

Comment:
Yes it totally make sense that it might not be convenient to wait for user input to terminate a silo. 
But if we terminate the program without user input, the user wouldn't be able to see exception printed to console , and they would get confused on why program ended. We didn't print the exception message to logs either. 

And ShutdownSilo() handles the case when silo didn't start up correctly. Look at OrleansHostWrapper.Stop(). In main method, in my opinion, for the goal of precise and readability, we should just express to people the simple logic: start the silo, wait for your input, then we shut down the silo. If people wants more detailed info, they can dig into one layer down to specific methods to know how those method handle exceptions, edge cases, etc